### PR TITLE
event-ingester: Add remote sbatch

### PR DESCRIPTION
**Pull request by Cocopyth**
## Description

- Added the possibility of launching a sbatch command on remote server after API ingestion and export.

- Set the time range of API call to 4 days before the day of execution

- Addapted `main.py` and `worker.py` to take care of both ingestion and sbatch command calling
- Makes other pull requests from "ssh" branches obsolete

## Implementation

- changed SFTPClient to more general SSHclient that can take care of both
- adapted `ingest.py` accordingly
- Added sbatch command to Settings (both in `config.py` and `compose.yml`)
- adapted `main.py` and `worker.py` to also take care of sbatch launching
- TODO: add `time_range` and `remote` to settings/config and ensure correct sbatch formating

### CODE REVIEW

File: export-ingester/export_ingester/config.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ class Settings(BaseSettings):
     SFTP_HOST: str
```
Comment (iw108) [inline export-ingester/export_ingester/config.py:12]: `TIME_RANGE` can be specified in the settings

File: export-ingester/export_ingester/main.py [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+from .utils import get_range
 
 
 async def main():
     settings = Settings()
-    remote = f"daily-uploads/{date.today()}.json"
+    remote = f"/scratch-shared/amftrack2024/daily/{date.today()}.json"
+    time_range = 4
+    # TODO these two need to become either parameter or environment variables
+    date_ = date.today() - timedelta(days=time_range)
```
Comment (iw108) [inline export-ingester/export_ingester/main.py]: is this correct? I think you're pulling data between 4 and 8 days ago. Is that what you want?

File: export-ingester/export_ingester/worker.py [hunk 3]
```diff
@@ -25,14 +25,15 @@ async def startup(ctx: dict):
 
 async def run_ingestion(ctx: dict, *, _date: date | None = None):
     settings: Settings = ctx["settings"]
-
-    date_ = _date or (date.today() - timedelta(days=1))
-    start, end = get_range(date_)
+    remote = f"/scratch-shared/amftrack2024/daily/{date.today()}.json"
+    time_range = 4
+    # TODO these two need to become either parameter or environment variables
+    date_ = _date or (date.today() - timedelta(days=time_range))
```
Comment (iw108) [inline export-ingester/export_ingester/worker.py]: See pervious comment

File: export-ingester/export_ingester/worker.py [hunk 4]
```diff
@@ -25,14 +25,15 @@ async def startup(ctx: dict):
 
 async def run_ingestion(ctx: dict, *, _date: date | None = None):
     settings: Settings = ctx["settings"]
-
-    date_ = _date or (date.today() - timedelta(days=1))
-    start, end = get_range(date_)
+    remote = f"/scratch-shared/amftrack2024/daily/{date.today()}.json"
```
Comment (iw108) [inline export-ingester/export_ingester/worker.py:28]: This could maybe also be configured in the settings?

File: export-ingester/poetry.lock
Comment (iw108) [inline export-ingester/poetry.lock:1]: Nothings been installed so you're lock file shouldn't be changing

File: export-ingester/export_ingester/ingest.py [hunk 5]
```diff
@@ -5,47 +5,54 @@
...
+        APIClient(
+            str(settings.BASE_URL),
+            http_client,
+        ),
+        ssh_client,
     )
 
     await stack.aclose()
+
+    await stack.aclose()
```
Comment (iw108) [inline export-ingester/export_ingester/ingest.py]: Second call isn't necessary

File: export-ingester/export_ingester/main.py (hunk 2 above)
Comment (Cocopyth) [reply export-ingester/export_ingester/main.py]: Oh then, I did get confused. I want everything between today and 4 days before. Fixed

File: export-ingester/export_ingester/config.py [hunk 6]
```diff
@@ -10,7 +10,8 @@ class Settings(BaseSettings):
...
 
     BASE_URL: HttpUrl
-
+    SBATCH_COMMAND: str
+    TIME_RANGE: str
```
Comment (iw108) [inline export-ingester/export_ingester/config.py]: I think you should maybe do 

```python
from datetime import timedelta

class Settings:
       ...

       TIME_RANGE: timedelta = timedelta(days=4)
```

File: export-ingester/export_ingester/main.py [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
 from .config import Settings
 from .ingest import get_managed_export_ingester
+from .utils import get_range
 
 
 async def main():
     settings = Settings()
-    remote = f"daily-uploads/{date.today()}.json"
+    remote = f"/scratch-shared/amftrack2024/daily/{date.today()}.json"
+    time_range = int(settings.TIME_RANGE)
```
Comment (iw108) [inline export-ingester/export_ingester/main.py]: if you type this in the settings you don't need to cast it here

File: export-ingester/export_ingester/main.py [hunk 8]
```diff
@@ -1,16 +1,22 @@
...
+    # TODO these two need to become either parameter or environment variables
+    date_ = date.today() - timedelta(days=1)
```
Comment (iw108) [inline export-ingester/export_ingester/main.py]:
```python
# probably in utils.py
def get_date_start(date_: date) -> datetime:
    return datetime(date_.year, date_.month, date_.day, tzinfo=UTC)


async def main():
      settings = Settings()

      start = get_date_start(date.today())
      end = start - settings.TIME_RANGE

```

File: export-ingester/export_ingester/utils.py [hunk 9]
```diff
@@ -5,8 +5,8 @@ def now() -> datetime:
     return datetime.now(tz=UTC)
 
 
-def get_range(date_: date) -> tuple[datetime, datetime]:
-    next_day = date_ + timedelta(days=1)
+def get_range(date_: date, time_range: int) -> tuple[datetime, datetime]:
```
Comment (iw108) [inline export-ingester/export_ingester/utils.py:8]: I think this complicates things a bit - see above suggestion re `get_date_start`

**Pull request merged**